### PR TITLE
feat(slides): add pagination-bottom property to the slider pagination

### LINKED ISSUE #15412: swiper-pagination-bullets position needs to be customizable
# Bug Report

**Ionic Info**
```
@ionic/core@4.0.0-beta.6
```

**Describe the Bug**
I have a full screen slider, when it appears on iPhone X, the pagination bullets are too close to the bottom. I'd like to move them up using `env(safe-area-inset-bottom)`. Note: If you don't have a full screen slider, you probably won't need to move the bullets.

**Steps to Reproduce**
1. Create slider
2. swiper-pagination-bullets are part of shadow dom so you can't style

**Expected Behavior**
css var available to customize bottom positioning of bullets.

**Additional Context**
image

### PR DESCRIPTION
**Pull request by notiv-nt**
#### Short description of what this resolves:
Closes #15412

#### Changes proposed in this pull request:

- Add css variable --pagination-bottom to the slider pagination

**Ionic Version**: 4.x

**Fixes**: #15412

### CODE REVIEW

File: core/src/components/slides/slides.scss [hunk 1]
```diff
@@ -55,3 +55,13 @@
...
   background: var(--scroll-bar-background-active);
 }
+
+
+// Pagination
+// Rewrite vendor styles
+// --------------------------------------------------
+.swiper-pagination-fraction,
+.swiper-pagination-custom,
+.swiper-container-horizontal > .swiper-pagination-bullets {
```
Comment (manucorporat) [inline core/src/components/slides/slides.scss]: Why not just `.swiper-pagination-bullets`?

File: core/src/components/slides/slides.md.vars.scss [hunk 2]
```diff
@@ -26,3 +26,6 @@ $slides-md-scroll-bar-drag-background-alpha:    .5 !default;
 
 /// @prop - Background color of the scroll bar drag handle
 $slides-md-scroll-bar-drag-background:          rgba(var(--ion-text-color-rgb, $text-color-rgb), $slides-md-scroll-bar-drag-background-alpha) !default;
+
+/// @prop - Value of the pagination bottom property
+$slides-md-pagination-bottom:                   10px !default;
```
Comment (manucorporat) [inline core/src/components/slides/slides.md.vars.scss]: I guess we should add full customization of the position, maybe adding top, left, right?

File: core/src/components/slides/slides.scss (hunk 1 above)
Comment (notiv-nt) [reply core/src/components/slides/slides.scss]: I took it from here https://github.com/ionic-team/ionic/blob/master/core/src/components/slides/slides-vendor.scss#L224
I dont sure, but i think vendor's selector, will rewrite `.swiper-pagination-bullets`

File: core/src/components/slides/slides.md.vars.scss (hunk 2 above)
Comment (notiv-nt) [reply core/src/components/slides/slides.md.vars.scss]: Yes, will do it later

**Comment by brandyscarney** [comment]
Thanks for the PR! Is this still something for us to look into merging? I am not able to see why the build isn't working. Is it possible to update this?

**Comment by notiv-nt** [comment]
@brandyscarney 
I promised to add more properties, but I stopped working with Ionic, so I forgot about it, if necessary, I will add

**Comment by stephannagel** [comment]
@mhartington Is this still on the cards?

**Comment by brandyscarney** [comment]
Thanks so much for the PR! I took another look into this and found there were some issues with how swiper sets the position for different pagination types. There are a few places where it sets a different position based on the type:

Horizontal Bullets: https://github.com/ionic-team/ionic/blob/master/core/src/components/slides/slides-vendor.scss#L224
Vertical Dynamic Bullets: https://github.com/ionic-team/ionic/blob/master/core/src/components/slides/slides-vendor.scss#L305
Horizontal Dynamic Bullets: https://github.com/ionic-team/ionic/blob/master/core/src/components/slides/slides-vendor.scss#L323
Progress Bar: https://github.com/ionic-team/ionic/blob/master/core/src/components/slides/slides-vendor.scss#L368
(and many more)

Due to the fact this can now be done with CSS & setting CSS variables inside of the pagination element itself would be the same amount of CSS selectors, I am going to close this. 

Please see my comment here for more info: https://github.com/ionic-team/ionic/issues/15412#issuecomment-551919083. Thank you!

**Pull request closed without merging**